### ISSUE TITLE
Accessibility: mybinder.org

### ISSUE THREAD

**Issue by Joy-Remison** [jupyterhub/outreachy#69]
### What page is this for?

https://mybinder.org/

### WAVE accessibility report

![Screenshot (55)](https://user-images.githubusercontent.com/103270155/196049704-ceee0bb0-1647-44e0-bd79-547bc1185fcd.png)

### The HTML element to be changed

```html
ELEMENTS ON BINDER THAT REQUIRE CHANGE;
1.Missing Alternative Text:
(A). <img id="logo" src="/static/logo.svg?v=fe52c40adc69454ba7536393f76ebd715e5fb75f5feafe16a27c47483eabf3311c14ed9fda905c49915d6dbf369ae68fb855a40dd05489a7b9542a9ee532e92b" width="390px">. 
(B).<img id="badge" src="/static/images/badge_logo.svg?v=51b20aa16836ea83f5ed69194c660eb85c4c2c1e32565312baedb7d534e3ffcf592881dbbe3da441d8293ded842755c906b91fb4aadf15220cf48111ebf701c4">

2.Language missing or invalid: The language of the document is not identified or a lang attribute value is invalid.
<html style="margin-left: 380px;">

3.Very low contrast: Very low contrast between text and background colors. currently, the contrast between the background is 3.85.1
(A).<a href="https://turing.ac.uk" style="color: rgb(51, 122, 183); background-color: rgb(235, 236, 237);">
Turing Institute
</a>

4.Missing first level heading: Headings facilitate page navigation for users of many assistive technologies. They are used by search engines to index the structure and content of the webpages. Headings show the proper structuring of a webpage and the presence of headings make it easier to navigate the webpage.

5.No page regions: This webpage lacks region Landmarks. Region landmarks make it easier to navigate the webpages and also provide structure and organization to the webpage with the right aesthetics and overall appearance.
```

### Your proposed HTML (or CSS), after change

```html
PROPOSAL OF WHAT THE HTML SHOULD BE;
1.Missing Alternative Text:
Proposed solution: Adding an alternative text is necessary to provide information about the Image displayed, especially for images that cannot be displayed due specific reasons for example Poor network, use of screen reader or an error in image source. So I added an alternative text to every Html code without an alternative text.
(A).<img id="logo" src="/static/logo.svg? v=fe52c40adc69454ba7536393f76ebd715e5fb75f5feafe16a27c47483eabf3311c14ed9fda905c49915d6dbf369ae68fb855a40dd05489a7b9542a9ee532e92b" alt="Binder logo Image" width="390px">.
(B).<img id="badge" src="/static/images/badge_logo.svg?v=51b20aa16836ea83f5ed69194c660eb85c4c2c1e32565312baedb7d534e3ffcf592881dbbe3da441d8293ded842755c906b91fb4aadf15220cf48111ebf701c4" alt="Image link to show binder badge snippet">.

2.Language missing or invalid:
Proposed solution: Adding the lang attribute value to the page to specify the language of the web content. This is to make the web page more accessible to users especially for readers who access the web page using screen readers, to enable the web browsers display the web characters and scripts properly, to enable media players and other third-party component display text correctly, to  enable the page user change fonts and make other adjustments with predictable results.
Also, adding the lang attribute value to the website will improve the website's conformance with the Web Content Accessibility Guidelines (WCAG).

<html lang="en" style="margin-left: 380px;">

3.Very low contrast: 
Proposed solution: Increasing the contrast between the foreground (text) color and the background color. In order to ensure that both elements have both foreground AND background colors defined. The contrast should be between 4.5-7.28 in order to provide sufficient contrast according to the WCAG requirements.
(A).<a href="https://turing.ac.uk" style="color: rgb(51, 122, 183); background-color: rgb(250, 236, 237);">
Turing Institute
</a>

4.Missing first level heading:
Proposed solution: I wound restructure the headings of the webpage according to hierarchy. Using <h1> for the main heading and gradually reducing it for the other headings in order to properly construct the web page making it easier to navigate the various sections of the page.

5.No page regions:
Proposed solution: I wound implement the ARIA technique by using more <sections> to divide the page into sections,<main> to different the main page from other pages, use of <footer>,<nav> to create a navbar and <header> to provide more organization to the webpage.
```

### More info

1. Adding an event listeners to the webpage such as mouseover, on click event ,getboundingClientRect to improve page accessibility, the focus and blur event to improve accessibility for key board users.
2. smooth scroll function to enable smooth scrolling effect of the webpage.
3. Adding a scroll button to enable the page user scroll back to the top of the page.

**Comment by minrk** [comment]
Thank you for the suggestions!

**Comment by sgibson91** [comment]
Thank you for your contribution! Since the contribution period is now over, we will close this issue.
